tests/tests.js

```diff
@@ -147,7 +148,23 @@ var runTests = function runTests(arrayFrom) {
 		t.end();
 	});
 
-	test('no setters are called for indexes', { 'skip': !Object.defineProperty }, function (t) {
+	var numericPropertySetterBug = (function () {
```

Comment:
we could make this slightly more efficient by adding `supportsDescriptors &&` in front of the test, but nbd